### PR TITLE
Added persistence

### PR DESCRIPTION
**Pull request by rak108**
with regards to #16

### CODE REVIEW

**Comment by krithikvaidya** [comment]
Test it after #23 is resolved, will merge after that

File: replicated_kv_store/raft_node.go
Comment (krithikvaidya) [inline replicated_kv_store/raft_node.go:100]: This needs to be done at the very beginning of the program, before the gRPC server is up (when the gRPC server is up it'll be listening for RPCs from other nodes, so the persisted data should be loaded before that)

File: replicated_kv_store/rpcs.go
Comment (krithikvaidya) [inline replicated_kv_store/rpcs.go]: There's no change to any of the data to be persisted at this point, so this might be redundant

File: replicated_kv_store/storage.go
Comment (krithikvaidya) [inline replicated_kv_store/storage.go]: value

File: replicated_kv_store/storage.go
Comment (krithikvaidya) [inline replicated_kv_store/storage.go]: value

File: replicated_kv_store/storage.go
Comment (krithikvaidya) [inline replicated_kv_store/storage.go]: os.Open(filename)

File: replicated_kv_store/raft_node.go
Comment (krithikvaidya) [inline replicated_kv_store/raft_node.go]: Get also takes the filename as param

File: replicated_kv_store/storage.go
Comment (krithikvaidya) [inline replicated_kv_store/storage.go:83]: The persistToStorage method should be called basically everywhere that any of the data items to be persisted have changed - so you'll also need to call it in the state transition functions

**Review by krithikvaidya — CHANGES_REQUESTED** [review]

File: replicated_kv_store/raft_node.go
Comment (rak108) [inline replicated_kv_store/raft_node.go:41]: Ignore this, this is for my new issue

File: replicated_kv_store/raft_node.go
Comment (rak108) [inline replicated_kv_store/raft_node.go:77]: ignore

File: replicated_kv_store/storage.go
Comment (rak108) [reply replicated_kv_store/storage.go:83]: This has been done

**Comment by krithikvaidya** [comment]
Resolve merge conflicts at the end

**Pull request merged**
